packages/components/card/src/card.tsx

```diff
@@ -25,12 +26,33 @@ export type TCardProps = {
    */
   className?: string;
   children?: ReactNode;
+  /**
+   * The callback function to be executed when the Card component is clicked. Prefer this for managing side effects rather than navigation.
+   */
+  onClick?: () => void;
+  /**
+   * The URL that the Card should point to. If provided, the Card will be rendered as an anchor element.
+   */
+  to?: string;
+  /**
+   * A flag to indicate if the Card points to an external source.
+   */
+  isExternal?: boolean;
+  /**
+   * Indicates that a clickable Card should not allow clicks. This allows consumers to temporarily disable a clickable Card.
+   */
+  isDisabled?: boolean;
 };
 
-const Card = (props: TCardProps) => (
-  <div
-    {...filterDataAttributes(props)}
-    css={css`
+const Card = (props: TCardProps) => {
+  const isClickable = Boolean(!props.isDisabled && (props.onClick || props.to));
+
+  const commonProps = {
+    ...filterDataAttributes(props),
+    onClick: isClickable ? props.onClick : undefined,
+    role: isClickable ? 'button' : undefined,
```

Comment:
In my opinion we should omit that, this would result in `<a role="button" ...`

Reply:
wouldn't this resolve to `<a role="button"...` only if `to` is passed? 
```suggestion
    role: isClickable && !to ? 'button' : undefined,
```

Reply:
`isClickable` is only `true` if the card is not disabled and we have either an `onClick` handler or `to` (which will resolve to an `<a>`)

`const isClickable = Boolean(!props.isDisabled && (props.onClick || props.to));`

Reply:
Wow, @ByronDWall, I need to learn to read - will make that change

Still waking up 😆

Reply:
@ByronDWall @kark please see dcca0ca60c27aa725343edcbc35f98e03a27c727, I think this should be correct

Reply:
❤️ Thanks, I think it is correct. However, now it seems to me we should also consider [addressing the accessibility aspect](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#:~:text=Note%3A%20If,example%20code.).

Reply:
>Note: If using role="button" instead of the semantic <button> or <input type="button"> elements, you will need to make the element focusable and define event handlers for [click](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) and [keydown](https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event) events. This includes handling the Enter and Space keypresses in order to process all forms of user input. See [the official WAI-ARIA example code](https://www.w3.org/WAI/ARIA/apg/patterns/button/examples/button/).

Are you referring to this portion?

Reply:
@FilPob Wondering if you can weigh in here, if we're going to support `button` accessibility for this component, we should have an idea of what a focus state should look like

Reply:
@FilPob this is the default focus state, just for your reference:
![image](https://github.com/commercetools/ui-kit/assets/15024562/7f27be6b-4ba5-4b82-8b44-e4ee7515d9e8)

Reply:
perfect, default is fine

Reply:
Thank you for all the improvements! ❤️ I don't think [this](https://github.com/commercetools/ui-kit/pull/2706#discussion_r1469767596) has been addressed. I'm still wondering if the `<div>` with `role="button"` should be accessible. Adding `tabIndex` of `0` and an `onKeyDown` prop in case the `<div>` is clickable should be enough.

Reply:
Thanks for that reminder, addressed in 3bac38bdbcb6b25458d7af61d66c1eefa7aed33b 👍